client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -138,31 +139,43 @@ export const receiveCommentError = ( { dispatch, getState }, { siteId, commentId
 
 // @see https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/comments/
 export const fetchCommentsList = ( { dispatch }, action ) => {
-	const { query } = action;
-
-	if ( ! query || 'site' !== query.listType ) {
+	if ( ! action.query || 'site' !== action.query.listType ) {
```

Comment:
Do you mean why we abort if `query` is provided but `query.listType !== 'site'`?
Or in other words, why do we even need `listType`, if we expect it to be one specific value, but then we don't use it anywhere?

I honestly have no idea.
Probably the idea here is to have this extendable to fetching either site or post comments, but I'm not sure.